src/devana/syntax_abstraction/functioninfo.py

```diff
@@ -291,7 +444,35 @@ def check_eq(a: FunctionInfo, b: FunctionInfo) -> bool:
     def modification(self) -> FunctionModification:
         """Function modification enum flag."""
         self._modification = FunctionModification.NONE
+
+        parse_noexcept: bool = False
+        parse_noexcept_value: bool = False
+        noexcept_value: str = ""
+        noexcept_has_value: bool = False
+
         for token in self._cursor.get_tokens():
+            if parse_noexcept:
+                if token.spelling == "(":
+                    parse_noexcept_value = True
+                    continue
+                elif parse_noexcept_value:
+                    if token.spelling == ")":
+                        if not noexcept_has_value:
+                            raise ParserError("xd")
+
+                        self._modification |= FunctionModification.NOEXCEPT(noexcept_value) # noqa pylint: disable=not-callable
+                        parse_noexcept = False
+                        parse_noexcept_value = False
+                    else:
+                        noexcept_has_value = True
+                        noexcept_value += token.spelling
+                    continue
+                else:
+                    self._modification |= FunctionModification.NOEXCEPT
+                    parse_noexcept = False
+                    parse_noexcept_value = False
```

Comment:
Please parse noexcept value inside "elif token.spelling == "noexcept":".
For now if no more tokens after noexcept this code will never invoke.